luarules/gadgets/lups_shield.lua

```diff
@@ -23,6 +23,7 @@ end
 local GAMESPEED = Game.gameSpeed
 local SHIELDARMORID = 4
 local SHIELDARMORIDALT = 0
+local SHIELDONRULESPARAMINDEX = 531313
```

Comment:
Rules params accept strings as keys, why not use those?

Reply:
Storing and referencing a number key will be more performant than a string if it's called repeatedly like it is here, correct? In which case I would prefer the more performant option as it appears a great deal of effort has gone into making this performant

Reply:
I expect it's a bit more performant but I don't think it has ever been measured so it's hard to tell by how much. I don't know what the guideline on perf vs maintainabiltity is so it's up to @WatchTheFort. If he agrees to keep it as a number then perhaps document it though:
```suggestion
local SHIELDONRULESPARAMINDEX = 531313 -- not a string due to perfmaxxing
```

Reply:
How on earth would referencing a number vs a string make any real, measurable performance difference?

Always favour code readability, other devs needs to be able to divine what everything means. What does this variable refer to?

Reply:
the index of a customrulesparam that's checked every update to determine if the shield bubble should be drawn or not.

Reply:
this appears both in the unit_shield_behavior.lua gadget and in the shield_lups.lua gadget, which are the majority of places shield code is handled game-side